packages/block-library/src/navigation-link/index.php

```diff
@@ -322,13 +322,32 @@ function build_variation_for_navigation_link( $entity, $kind ) {
 	return $variation;
 }
 
+/**
+ * Register a variation for a post type / taxonomy for the navigation link block
+ *
+ * @param array $variation Variation array from build_variation_for_navigation_link.
+ * @return void
+ */
+function register_block_core_navigation_link_variation( $variation ) {
+	// Directly set the variations on the registered block type
+	// because there's no server side registration for variations (see #47170).
+	$navigation_block_type = WP_Block_Type_Registry::get_instance()->get_registered( 'core/navigation-link' );
+	if ( ! $navigation_block_type ) {
+		return;
+	}
+
+	$navigation_block_type->variations[] = $variation;
+}
+
 /**
  * Register the navigation link block.
  *
  * @uses render_block_core_navigation()
  * @throws WP_Error An WP_Error exception parsing the block definition.
  */
 function register_block_core_navigation_link() {
+	// This will only handle post types and taxonomies registered until this point (init on priority 9).
+	// See action hooks at the end of this function for other post types and taxonomies.
```

Comment:
```suggestion
	// This will only handle post types and taxonomies registered until this point (init on priority 9).
	// See action hooks at the end of this function for other post types and taxonomies.
	// See https://github.com/WordPress/gutenberg/issues/53826 for details.
```